examples/common/tracing/TracingCommandLineArgument.cpp

```diff
@@ -36,6 +43,28 @@ using ::chip::Tracing::LogJson::LogJsonBackend;
 // currently supported backends
 LogJsonBackend log_json_backend;
 
+#ifdef ENABLE_PERFETTO_TRACING
+
+using ::chip::Tracing::Perfetto::FileTraceOutput;
+using ::chip::Tracing::Perfetto::PerfettoBackend;
+
+FileTraceOutput perfetto_file_output;
+PerfettoBackend perfetto_backend;
+
+bool StartsWith(CharSpan argument, const char * prefix)
+{
+    const size_t prefix_len = strlen(prefix);
+    if (argument.size() < prefix_len)
+    {
+        return false;
+    }
+
+    argument.reduce_size(prefix_len);
+    return argument.data_equal(CharSpan(prefix, prefix_len));
+}
+
+#endif
+
 // ScopedRegistration ensures register/unregister is met, as long
 // as the vector is cleared (and we do so when stopping tracing).
 std::vector<std::unique_ptr<ScopedRegistration>> tracing_backends;
```

Comment:
Can this be owned by the program, directly or indirectly?

Having a global vector is problematic because we end up with unpredictable atexit() destruction (and it therefore isn't allowed by Google style guides).

Reply:
Removed the vector.

Having program ownership would require app-main to be more class-oriented. I think that change should be some separate PR since it may involve more program changes (app main is used by all examples).